src/config.rs

```diff
@@ -70,55 +74,72 @@ pub fn init_sys_cfg() -> Result<SysCfg> {
 
     cfg_file.read_to_string(&mut cfg_contents)?;
 
-    trace!("Overwrite existing file to preserve consistency");
-    cfg_file = OpenOptions::new()
-        .write(true)
-        .truncate(true)
-        .open(&ENV_CFG.data_cfg_file)?;
-
     let sys_cfg: SysCfgFile = toml::from_str(&cfg_contents)?;
 
+    let http_port = sys_cfg.http_port.unwrap_or(7000);
+
     let private_key = sys_cfg
         .private_key
         .unwrap_or_else(|| SecretKey::new(&mut rand::thread_rng()));
 
-    let volumes: Vec<Volume> = sys_cfg
-        .volumes
-        .map(|vols| {
-            vols.iter()
-                .map(|vol| Volume {
-                    path: PathBuf::from(&vol.path),
-                })
-                .collect()
-        })
-        .unwrap_or_else(|| {
-            (0..8)
-                .map(|i| Volume {
-                    path: PathBuf::from(format!("/tmp/carbonado-{i}")),
-                })
-                .collect()
+    let secp = Secp256k1::new();
+    info!("Node Public Key: {}", private_key.public_key(&secp));
+
+    let drive_redundancy = sys_cfg.drive_redundancy.unwrap_or(1);
+
+    match drive_redundancy {
+        1 | 2 | 4 | 8 => {}
+        _ => {
+            return Err(anyhow!("drive_redundancy must be either 1, 2, 4, or 8"));
+        }
+    }
+
+    let orig_volumes = sys_cfg.volumes.unwrap_or_default();
+    let mut volumes: Vec<Volume> = vec![];
+    let base_dirs = BaseDirs::new();
+    let base_dir = match base_dirs {
+        Some(base_dir) => base_dir.home_dir().to_owned(),
+        None => PathBuf::from("/tmp"),
+    };
+
+    for i in 0..drive_redundancy {
+        let volume: Volume = orig_volumes.get(i).map(|v| v.to_owned()).unwrap_or(Volume {
+            path: base_dir.join("Carbonado").join(format!("Volume_{i}")),
         });
 
+        volumes.push(volume.to_owned());
+    }
+
     for vol in volumes.iter() {
-        create_dir_all(&vol.path.join(SEGMENT_DIR))?;
-        create_dir_all(&vol.path.join(CATALOG_DIR))?;
+        let segment_dir = vol.path.join(SEGMENT_DIR);
```

Comment:
I don't see a gain for this...
It looks as though you get the same result.

Reply:
You're probably right. I forget why I did this.